R/principalcomponentanalysis.R

```diff
@@ -137,13 +137,17 @@ PrincipalComponentAnalysis <- function(jaspResults, dataset, options, ...) {
       }
     })
   }
+  if (options[["basedOn"]] == "correlation") baseOn <- "cor"
+  else if (options[["basedOn"]] == "covariance") baseOn <- "cov"
+  else if (options[["basedOn"]] == "mixedCorrelationMatrix") baseOn <- "mixed"
```

Comment:
A little bit more readable approach would be like
```suggestion
baseOn <- switch(
  options[["basedOn"]],
  correlation            = "cor",
  covariance             = "cov",
  mixedCorrelationMatrix = "mixed"
  )
```